### PR TITLE
Adhoc :  Add PR template

### PR DESCRIPTION
**Pull request by gauravchaudhari02**
Signed-off-by: Gaurav Chaudhari <gaurav.chaudhari@seagate.com>

-----
[View rendered .github/pull_request_template.md](https://github.com/gauravchaudhari02/cortx-hare/blob/add-template/.github/pull_request_template.md)

### CODE REVIEW

**Comment by knekrasov** [comment]
I'm not sure that this template fits the needs of our components. I remember there was a discussion at a confluence page but I don't think that Hare agreed to use that template.

cc @mssawant

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Testing 
+  Checklist for Author
+- [ ] Unit and System Tests are added
+- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
+- [ ] Testing was performed with RPM
+
+# Impact Analysis
+  Checklist for Author/Reviewer/GateKeeper
+- [ ] Interface change (if any) are documented
+- [ ] Side effects on other features (deployment/upgrade)
```
[the remark this answers is not in the document]
Comment (knekrasov) [reply .github/pull_request_template.md:21]: What do we call an interface here, BTW?

I can guess that it is implicitly assumed that the component has its borders and some (clearly defined) public interfaces that this component integrates with others. The problem is that such interfaces have never been described.

If they are not described, how can developer or reviewer be sure that such interface is changed?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+-  For Feature, Post the link for design
+
+# Coding
+   Checklist for Author
+-  [ ] Coding conventions are followed and code is consistent
+
+# Testing 
+  Checklist for Author
+- [ ] Unit and System Tests are added
+- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
```
[the remark this answers is not in the document]
Comment (knekrasov) [reply .github/pull_request_template.md:15]: Without design, the notions of Happy path and Non-Happy path make no sense.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] Testing was performed with RPM
+
+# Impact Analysis
+  Checklist for Author/Reviewer/GateKeeper
+- [ ] Interface change (if any) are documented
+- [ ] Side effects on other features (deployment/upgrade)
+- [ ] Dependencies on other component(s)
```
Comment (knekrasov) [inline .github/pull_request_template.md:22]: Instead of yes/no, textual details are needed.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+# Review Checklist 
+  Checklist for Author
+- [ ] JIRA number/GitHub Issue added to PR
+- [ ] PR is self reviewed
```
Comment (knekrasov) [inline .github/pull_request_template.md:27]: What does it mean?

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] Jira and state/status is updated and JIRA is updated with PR link
```
Comment (knekrasov) [inline .github/pull_request_template.md:28]: `s/Jira/JIRA/`
Let's keep the naming consistent within a single document.

Comment (knekrasov) [reply .github/pull_request_template.md:28]: The word update doesn't tell what exactly is expected to happen to JIRA ticket. Perhaps a link to the document is required telling how PR workflow maps to JIRA ticket workflow.

**Comment by sunilsavanur** [comment]
> The aims of the template are not clear.

Objective of PR checklist is to bring hygiene factor or uniformity across all CORTX PRs to start with. In future team plans to analyze PR data for improvements.

**Comment by mssawant** [comment]
As discussed, team is going ahead with this initial version of the PR template.

**Pull request merged**